### ISSUE TITLE
VAT settings clarification

### ISSUE THREAD

**Issue by aswiniimpro** [allegro/allegro-api#4886]
Hi,

  We need some clarification about VAT settings.
  
  If the customer place an order, the VAT amount not showing at checkout to the customer. It shows only in settlements with allegro page of seller account.
  
  How the customer knows about VAT amount ? VAT Amount will be collected from buyer or seller ?
  
  I've purchased 2 orders , one using my normal sandbox account. another using my another business sandbox account's product.
  
  For poland 23% VAT, For belgium 19% VAT is showing in settlements page.
  
  For all offers inside my business account I have set the option as 'VAT Invoice' for offer creation.
  
  Where do we set the Other countries VAT Rate inside our account or  it is a standard VAT Rates defined in Allegro by default ?
    
  How do we identify if the order's customer account is a B2B or B2C ?
  
  I have used this endpoint to fetch orders from allegro. - https://api.allegro.pl.allegrosandbox.pl/order/checkout-forms
  
  In response, I got the item price, paid amount and delivery charges. I need to get VAT price related to that order also.
  
  Please let us know any other end point to get settlements details available in allegro and how VAT calculation works in allegro orders.

**Comment by AureliuszBrussy** [comment]
You can get all VAT settings available in the given category by using [GET /sale/tax-settings?category.id={categoryId}](https://developer.allegro.pl/en/documentation/#operation/getTaxSettingsForCategory). You can apply to offer only one of the available settings. You can read more about tax setting in our guides:
- [Invoice settings and tax rate](https://developer.allegro.pl/en/sale/#9-invoice-settings-and-tax-rate),
- [How to set the VAT invoice rate](https://developer.allegro.pl/en/product_offers/#how-to-set-the-vat-invoice-rate).


You can identify if orders are B2B in **invoice.required** field when getting details of the order via [GET /order/checkout-forms/{checkoutFormId}](https://developer.allegro.pl/en/documentation/#operation/getOrdersDetailsUsingGET).

You can get tax rate only by checking offers in which the buyer placed an order ([GET /sale/offers/{offerId}](https://developer.allegro.pl/en/documentation/#operation/getOfferUsingGET)).

Please, read more about [VAT e-commerce package](https://allegro.pl/help/news/on-july-1-the-vat-e-commerce-package-comes-into-force-aMRv33qKvCx). In some cases VAT can be accounted for by trading platforms, such as Allegro ― instead of the seller. You can read about adapting Allegro API to the VAT e-commerce package in [our news](https://developer.allegro.pl/en/news/#vat-2021).

If you do not know which VAT rate applies to your offers - check it in the [European VAT rates database](https://ec.europa.eu/taxation_customs/tedb/vatSearchForm.html) or contact your tax advisor.

**Comment by aswiniimpro** [comment]
Dear Team, Thanks for your response.

I gone through all  links you have sent.

"You can identify if orders are B2B in invoice.required field when getting details of the order via GET /order/checkout-forms/{checkoutFormId}."

- I have placed orders using common sandbox account type.  But, if i choose the option to get invoice for order while checkout, we got invoice.required field as true. But for that order we had VAT e-commerce charge. 

- So how do we conclude if the customer charged  for VAT or not using this field ?

I have **common** allegro sandbox account type. Where did i mention my VAT number inside my account ?

**Comment by aswiniimpro** [comment]
"You can get all VAT settings available in the given category by using GET /sale/tax-settings?category.id={categoryId}"

- I have checked this API response for category id 260341 and  260338. Both returns Goods VAT rate as 23%.

- I have purchases offers which are created using products assigned for these categories. 

- For Offer having 260341 category,  VAT percentage is 18.7
- For Offer having 260338 category,  VAT percentage is 19

- How do i know the VAT percentage for all offers ? We didn't give any specific VAT rate for offers.

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy , Can you please clarify my queries ?

**Comment by AureliuszBrussy** [comment]
> I have placed orders using common sandbox account type. But, if i choose the option to get invoice for order while checkout, we got invoice.required field as true. But for that order we had VAT e-commerce charge. - https://nimb.ws/LDdt7U
> 
> So how do we conclude if the customer charged for VAT or not using this field ?

Sorry for incomplete information - the condition for recognizing a B2B transaction is:
- willingness to receive a VAT invoice by the buyer (**invoice.required = true**),
- the buyer must complete the company details in the **invoice.company** (**name** and **taxId**) field.

> I have common allegro sandbox account type. Where did i mention my VAT number inside my account ?

As a Buyer, you can declare your tax ID at checkout form:
![faktura](https://user-images.githubusercontent.com/68948612/131795854-2e76963f-72d0-48ed-8344-931081705ad0.PNG)

after clicking on "Zmień dane" (Change data) you can add new data, and by selecting "Company" you can add tax ID and company name. 

> How do i know the VAT percentage for all offers ? We didn't give any specific VAT rate for offers.

You can check the declared VAT rate in a given offer only with [GET /sale/offers/{offerId}](https://developer.allegro.pl/en/documentation/#operation/getOfferUsingGET) or [GET /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation/#operation/getProductOffer), and only if you added VAT rate settings when creating or editing the offer.

Also: you do not have to declare VAT rates if the standard VAT rate (23%) applies to your offers in Poland, and you sell goods (instead of services). If you do not declare VAT rates and provide additional parameters for purposes of the VAT e-Commerce Package, we will apply the standard VAT rate. The buyers on Allegro Biznes will see gross prices ― just like now.
You need to declare a VAT rate if you find a reduced VAT rate (5% or 8%) applicable to your offers, or your sales are exempt from VAT.

If you have specific questions about the e-Commerce VAT package and Allegro settlements (for example, a question about a specific VAT rate charged in settlements with Allegro) - please use the [contact form](https://allegro.pl/help/contact?subjectId=86ad5dc6-1add-449a-9a00-522895fa81ff&category=753e61e1-6a7a-4c6c-b068-ae7587023c7c).

**Comment by aswiniimpro** [comment]
After clicking on "Zmień dane" (Change data) you can add new data, and by selecting "Company" you can add tax ID and company name.

- We can't have this option at checkout page. - https://nimb.ws/nDaMyL

can you please let us know.

**Comment by AureliuszBrussy** [comment]
@aswiniimpro you need to mark "I want to receive an invoice", then "Change data" ("Zmień dane") on the invoice address.

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy ,

I have contact about VAT % via the Allegro Contact form. They replied VAT % for the Poland is 23% and commission amount will vary depend upon offer category.

But I don't get 23% VAT inside settlements with allegro. https://allegro.pl.allegrosandbox.pl/my-allegro/my-account/billing/settlements-with-allegro

As i said before, we have 18.7% or 19% depends on category.

**Total:** 63.90 PLN
**Commision:** 6.08 PLN
**VAT Ecommerce Charges :** 11.95 PLN (18.7% of 63.90)

Can you please let me know how this VAT calculation works even though we have 23% for Poland.

**Comment by AureliuszBrussy** [comment]
Can you give me the login of the account on which this transaction took place (on sandbox)? I'll check the reason.

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy , We can give screen shot for any settings page you wish to check.

Can you please let us know that settings, we will send screen shot for that.

**Comment by AureliuszBrussy** [comment]
@aswiniimpro please, simply send me the username or email address of your sandbox account - this will allow me to verify why you have been charged this rate.

**Comment by aswiniimpro** [comment]
**Username:** improgroup

**Comment by AureliuszBrussy** [comment]
@aswiniimpro we treat the offer prices as gross, i.e. we do not add the VAT, but we calculate VAT from the gross amount.

In order to better explain this, I will use an example:
suppose that the price of an item is 123 PLN. The VAT on such an item is 23% - 100 PLN is the net price. Allegro charges 23 PLN from the seller, and the buyer pays 123 PLN. 23 PLN of 123 PLN is just about 18.7%.

**Comment by aswiniimpro** [comment]
@AureliuszBrussy Thanks for your reply.

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy 

We don't have state param value in address data from allegro. But we need this state value.
Why the allegro address form not having state field? Is any possibility to add state in address form ?

**Sample Response:**

"invoice": {
                "required": true,
                "address": {
                    "street": "xxxx",
                    "city": "yyy",
                    "zipCode": "zzzz zzz",
                    "countryCode": "PL",
                    "company": {
                        "name": "Test",
                        "taxId": "AAAAAAAAAA"
  },

**Comment by AureliuszBrussy** [comment]
We do not support a field such as "state", both in the API and in the form in which the buyer completes the delivery or invoice address. If you have a suggestion regarding data that in your opinion should be included - please report it via the [contact form](https://allegro.pl/pomoc/kontakt?kategoria=4a6df9f8-0556-4b18-8930-68906a58c18e&subjectId=6093913a-efdf-4b94-a444-bde85b8c3d8b).

**Comment by aswiniimpro** [comment]
Okay. Thank you.

**Comment by ImproROD** [comment]
@AureliuszBrussy - Please also let us know if Allegro will be the issuer of invoices to customers, as they are a marketplace deemed supplier under new EU legislation, just like Amazon, eBay and others.  

Please advice

thanks

ROD

For Impro Group

**Comment by AureliuszBrussy** [comment]
@ImproROD your question is not related to the Allegro API. Please, use the [contact form](https://allegro.pl/pomoc/kontakt?kategoria=4a6df9f8-0556-4b18-8930-68906a58c18e&subjectId=6093913a-efdf-4b94-a444-bde85b8c3d8b) - you will receive an answer from the relevant team.

**Comment by ImproROD** [comment]
@AureliuszBrussy : Let me rephrase it.  Are we expected to get data from the API to raise direct customer invoices ourselves or not?

**Comment by AureliuszBrussy** [comment]
@ImproROD your question touches on the issue of tax billing and the necessity (or not) of a VAT invoice. Perhaps one of [these articles](https://allegro.pl/help/search?phrase=e-commerce) in Allegro Help Center will help clarify this issue for you. If not - please use our [contact form](https://allegro.pl/help/contact?subjectId=86ad5dc6-1add-449a-9a00-522895fa81ff&category=753e61e1-6a7a-4c6c-b068-ae7587023c7c).

From the API perspective - if there is a need, you can get data from the Allegro API to issue a VAT invoice to the customer. You can get this data by using [GET /order/checkout-forms/{checkoutFormId}](https://developer.allegro.pl/en/documentation/#operation/getOrdersDetailsUsingGET). You can even use our API to add an invoice to order - you can read more in [our guide](https://developer.allegro.pl/en/orders/#invoices).

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy ,

  I have asked about sending Invoices to buyers who request invoices for their orders. 

They replied:
  
**We do not issue invoices for buyers for the B2C transactions. Since the Seller is the party to such a transaction, they may decide to issue other, non-tax documents to confirm such transactions for buyers.

You can attach the document in the Orders tab.**

- I have checked the invoice attachment option in seller portal. Found the filter option only. Can u please let me know the option to send invoice to buyers ?

- As we get encrypted email id from allegro api response, how the seller knows the buyer original mail id ?

- GET /order/checkout-forms/{checkoutFormId} - This is the API to get details about particular order along with invoice address and VAT details of buyer. Is any API available to send invoices documents to buyer ?

Please let us clarify these queries.

**Comment by AureliuszBrussy** [comment]
Hi @aswiniimpro 

1. Adding an invoice is possible only for orders where the customer has requested it. In [Order tab](https://allegro.pl/my-allegro/sales/orders/) you just need to click on 
![pasted image 0](https://user-images.githubusercontent.com/68948612/134330695-cce33eeb-fade-447b-87a9-20821547455c.png) icon, next to your order.

2. We display email addresses in the correspondence within Allegro only as a character series in the @allegromail.pl domain, for example: f8awgqyk6a5@allegromail.pl. You will not receive original email address. You can read more about it [here](https://allegro.pl/help/for-sellers/communicating-with-buyers/how-we-display-email-addresses-in-the-correspondence-within-allegro-VE2Zb0reBuO).

3. It is possible to add an invoice to order via our API. Check [our guide](https://developer.allegro.pl/en/orders/#invoices).

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy 

You will add an invoice to the order in two steps - first, use POST /order/checkout-forms/{id}/invoices to create invoice object.

- If we add invoice using this api, it will automatically sent to customer ?

**Comment by AureliuszBrussy** [comment]
@aswiniimpro yes, the buyer will receive an e-mail notification and will be able to download the invoice in the "My purchases" tab.

**Comment by aswiniimpro** [comment]
Thank you @AureliuszBrussy . We will check this API and confirm this work flow.

**Comment by aswiniimpro** [comment]
> @aswiniimpro yes, the buyer will receive an e-mail notification and will be able to download the invoice in the "My purchases" tab.

Yes, the buyer will receive notification email along with invoice document attachment.

I have searched the option to download invoice inside buyer account. But not found.
Allegro -> My Allegro -> Shopping -> My purchases and auctions -> My shopping

Can you please let us know the option ?

**Comment by AureliuszBrussy** [comment]
As a buyer, you should click on "Szczegóły zakupu" (purchase details). In the next window click on "Rozwiń szczegóły zakupu" (expand purchase details). Under "Dane do faktury" (invoice details) you will find the button "pobierz fakturę" (download invoice).

**Comment by aswiniimpro** [comment]
Thanks for your reply. Found the download option.

**Comment by aswiniimpro** [comment]
Hi @AureliuszBrussy ,

We are getting this response for offer creation and all api calls.

{"errors":[{"code":"ConstraintViolationException.ValidSeller","message":"You don't have the required permissions to publish offer (sell blocked). Verify your account as a company.","details":"ConstraintViolationException.ValidSeller","path":"seller","userMessage":"Nie masz wymaganych uprawnień, aby wystawić ofertę (sprzedaż zablokowana). Zweryfikuj swoje konto jako firmowe."}]}

Till yestreday all api calls works fine. Please let us know the reason and how to solve this.

**Comment by PrzemyslawLukanowski** [comment]
@aswiniimpro  I answered in the #4762 thread.

**Comment by aswiniimpro** [comment]
Hi @PrzemyslawLukanowski 

Recently we have initiated the VAT refund via allegro API call for our live orders. We got the status as 'INPROGRESS'. https://nimb.ws/fi4NO7

Later the refund status changed to 'NOT REFUNDED' - https://nimb.ws/h0nFPW

How do we know why the refund was failed ?

Order payment date: Aug 10, 2023
Refund Date: Aug 18, 2023

**API response we got for one of the order's VAT refund:**


{
  "id": "478c49cd-3dc2-11ee-92ce-19667999421b",
  "payment": {
    "id": "de4b9c75-3781-11ee-be69-1313a9f51293"
  },
  "reason": "REFUND",
  "status": "NEW",
  "createdAt": "2023-08-18T12:25:13.014Z",
  "totalValue": {
    "amount": "123.35",
    "currency": "PLN"
  },
  "lineItems": [
    {
      "id": "d8f86390-3781-11ee-bf19-31b9d6a903c7",
      "type": "AMOUNT",
      "value": {
        "amount": "123.35",
        "currency": "PLN"
      }
    }
  ],
  "surcharges": [],
  "sellerComment": "Sprawdziliśmy Twój polski numer VAT pod kątem sprzedaży transgranicznej w UE. 23% podlega zwrotowi zgodnie z unijnymi regulacjami IOSS 2021. Więcej informacji tutaj: https://taxation-customs.ec.europa.eu/ioss_en "
}


Please let us know the reason.

**Comment by PrzemyslawLukanowski** [comment]
In this case, we were not able to charge the funds from the seller's account within 24 hours. The refund was not ordered to the operator.
